modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -949,11 +980,34 @@ else if (!FAILED.equals(workflow.getState())) {
     return processingOperation;
   }
 
+  private boolean userCanAccessWorkflow(WorkflowDefinition wd) {
+    if (wd.getRoles().isEmpty() || securityService.getUser().getRoles().stream()
+            .anyMatch(role -> role.getName().equals(GLOBAL_ADMIN_ROLE))) {
+      return true;
+    }
+    final Set<Role> userRoles = securityService.getUser().getRoles();
+    final Set<Role> workflowRoles = wd.getRoles().stream()
+            .map(roleStr -> roleDirectoryService.getRolesStream().filter(role -> role.getName().equals(roleStr))
```

Comment:
What is the role for the RoleDirectoryService here? At the end of the day, the only thing this code needs to do is to check whether usersRoles contains at least one of the roles in wd.getRoles()...?

Reply:
Yeah, that's actually true. I just thought I maybe shouldn't rely on string comparisons but instead try to actually get an existing `Rule` object. If that's not necessary, great, I'll simplify the code. :)